api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/summary/FormatUtils.kt

```diff
@@ -0,0 +1,261 @@
+    }
+
+    require(rows.all { row ->
+        row.lines().all { it.size == columnNames.size }
+    }) { "All lines must have the same number of columns" }
+
+    return columnNames.indices.map { columnIndex ->
+        val column = rows.map { row -> row.lines() }.flatten().map { it[columnIndex] }
+        calcColumnWidth(columnNames[columnIndex].length, column)
+    }.toMutableList()
+}
+
+/**
+ * Format list of strings to a single line with appropriate paddings for each column.
+ * @param [columnSeparator] sequence of symbols to separate columns
+ * @param [columnsWidths] widths of all columns
+ * @param [strs] list of strings to substitute to columns
+ */
+public fun formatLine(columnSeparator: String, columnsWidths: List<Int>, strs: List<String>): String {
+    return columnsWidths
+        .mapIndexed { index, columnWidth -> (strs.getOrNull(index) ?: "").padEnd(columnWidth) }
+        .joinToString(separator = columnSeparator)
+}
+
+
+/**
+ * Pretty print table from multiple sections.
+ * Each section consist of multiple rows, each row may consist of multiple lines.
+ * Sections may or may not have columns.
+ *
+ * @see [Section]
+ */
+public fun formatTable(vararg sections: Section): List<String> {
+    require(sections.isNotEmpty()) { "At least one section must be provided" }
+
+    val result = mutableListOf<String>()
+
+    val sectionsWithColumns = sections.filterIsInstance<SectionWithColumns>()
+
+    for (section in sectionsWithColumns) {
+        section.alignWith(sectionsWithColumns)
+    }
+
+    val simpleSections = sections.filterIsInstance<SimpleSection>()
+
+    for (section in simpleSections) {
+        section.alignWith(sectionsWithColumns)
+    }
+
+    for (section in sections) {
+        result.addAll(section.format())
+    }
+
+    return result;
+}
+
+
+/**
+ * Base interface for pretty print part of the model summary table.
+ */
+public sealed interface Section {
+    /**
+     * Format section to a list of strings.
+     */
+    public fun format(): List<String> = format(getWidth())
+
+    /**
+     * Format section to a list of strings with the specified width.
+     */
+    public fun format(width: Int): List<String>
+
+    /**
+     * Get width of the section.
+     */
+    public fun getWidth(): Int
+
+    /**
+     * Align section with the another section to have all corresponding columns with the same width.
+     */
+    public fun alignWith(anotherSection: Section)
+
+    /**
+     * Align section with the another sections
+     * to have all columns equal to the maximum width of the corresponding column of the another sections.
+     */
+    public fun alignWith(anotherSections: List<Section>)
+}
+
+
+/**
+ * Section with columns of the model summary table.
+ * Usually it is a header or footer of the table.
+ */
+public class SimpleSection(
+    private val rows: List<TableRow>,
+    private val lineSeparatorSymbol: Char,
```

Comment:
Do we want separator symbols as a part of a model? They seem to be more related to the presentation. So we can do something like `section.format(lineSeparatorSymbol = "*")` (passing different separators to format the same section).